packages/cli/src/lib/packager/packageManager.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { join as joinPath } from 'path';
+import { existsSync } from 'fs';
+import { findRootSync } from '@manypkg/find-root';
+
+interface PackageManager {
+  command: string;
+  lockfile: string;
+}
+
+function detect(): PackageManager {
+  const rootDir = findRootSync(process.cwd());
+  const lockfiles = {
+    'yarn.lock': 'yarn',
+    'package-lock.json': 'npm',
+  };
+
+  for (const [lockfile, command] of Object.entries(lockfiles)) {
+    if (existsSync(joinPath(rootDir, lockfile))) {
+      if (command !== 'yarn')
```

Comment:
The comment is likely to be useful to all developers, including existing `yarn` developers. While I acknowledge this change may be seeking entirely backwards-compatible log changes to avoid a new entry, I think this change warrants the new log entry.

Reply:
changed.